Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/GetPageMainPage/GetPageOfStreetcodesMainPageHandler.cs

```diff
@@ -45,7 +46,15 @@ public async Task<Result<IEnumerable<StreetcodeMainPageDTO>>> Handle(GetPageOfSt
 
                 var streetcodesPaginated = streetcodes.Paginate(request.page, request.pageSize);
 
-                return Result.Ok(_mapper.Map<IEnumerable<StreetcodeMainPageDTO>>(streetcodesPaginated));
+                var shuffledStreetcodes = streetcodes.OrderBy(sc =>
+                {
+                    using var rng = RandomNumberGenerator.Create();
```

Comment:
It will be more efficient to create a RandomNumberGenerator only once
Like this 

 var shuffledStreetcodes = new List<StreetcodeContent>();

 using (var rng = RandomNumberGenerator.Create())
 {
     shuffledStreetcodes = streetcodes.OrderBy(sc =>
     {
         byte[] random = new byte[4];
         rng.GetBytes(random);
         return BitConverter.ToInt32(random, 0) & 0x7FFFFFFF;
     }).ToList();
 }
 
 You can also do something like this

var shuffledStreetcodes = streetcodes.OrderBy(sc => rng.Next());
It looks much simpler

Reply:
Noted! Thx

Reply:
Also, "Next()" is not implemented inside of RandomNumberGenerator, it is tho implemented in Random, BUUUT, it raises another problem with cryptography, because it uses time dependent seed